ml_genn/model.py

```diff
@@ -237,6 +237,7 @@ def evaluate(self, data, labels, time, save_samples=[]):
                 for output_i in range(len(self.outputs)):
                     predictions = self.outputs[output_i].neurons.get_predictions(
                         pipe_batch_end - pipe_batch_start)
+                    batch_labels[output_i] = [np.argmax(i) for i in batch_labels[output_i]] if batch_labels[output_i].shape != predictions.shape else batch_labels[output_i]    
```

Comment:
Looks good! My only comment would be that this is an overly long and overly complex line of code. This would be much clearer:
```python
if batch_labels[output_i].shape != predictions.shape:
     batch_labels[output_i] = [np.argmax(i) for i in batch_labels[output_i]]
```